docs/guides/src/main/server/outgoinghttp.adoc

```diff
@@ -0,0 +1,147 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+
+<@tmpl.guide
+title="Configure outgoing HTTP requests"
+summary="How to configure Keycloak's HTTP client for outgoing requests."
+includedOptions="">
+
+Keycloak often needs to make requests to the applications and services it secures. Keycloak manages these outgoing connections using an HTTP client. This guide shows how to configure the client, it's connection pool, proxy environment settings, timeouts and more.
+
+== Usage
+To configure Keycloak's outgoing Http client, use:
+
+<@kc.start parameters="--spi-connections-http-client-default-<configurationoption>=<value>"/>
+
+Please see <<Outgoing Http Client Configuration Options>> for the available configuration options.
+
+== Outgoing Http Client Configuration Options
+
+The Http client Keycloak uses for outgoing communication is highly configurable. The available options are:
+
+*establish-connection-timeout-millis*::
+Maximum time in milliseconds until a connection approach times out. Default: Not set.
```

Comment:
```suggestion
Maximum time in milliseconds until a establishing a connection times out. Default: Not set.
```

Reply:
Hi,
Thanks for updating this part of documentation. :+1: 
Last week we had a production problem while calling (asynchronously) an external API and we noticed in thread dumps that problem was more in establishing connection (handshake) rather than waiting for response. 

```
- locked <0x00000006659d21b0> (a sun.security.ssl.TransportContext)
at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436)
```

So we were wondering : is there any reason that `establish-connection-timeout-millis` does not have a default value, like a few seconds ? 
No default value  (-1)  means that establishing connection can last forever, don't you think that it might cause some http calls to wait forever if a network problem occurs _before connection has been established_ ?

Reply:
@daviddelannoy sorry, this comment slipped through. would you mind opening a discussion or an issue for this? I'm with you this is not a good default, but I don't have all the context so might be worth to ask first why it's done that way. Would also be nice if you could suggest a better default :)